### PR TITLE
Fix: Release job for Windows

### PR DESCRIPTION
**Pull request by thekaveman**
There were a couple of problems making the Release job fail on the Windows runner, which this PR fixes:

1. Ensure `bash` is used across all runners. Before, Windows used its default powershell, and so couldn't run the `bin/` scripts.
2. Fix the tests to be more newline agnostic. Before, the tests and sample data files assumed a `\n` newline.

Here's a manual run that only performed the first steps (e.g. the tests and packaging, not publishing a release), showing successful completion for all platforms: https://github.com/cal-itp/hashfields/actions/runs/5945152569

### CODE REVIEW

File: tests/test_hashfields.py [hunk 1]
```diff
@@ -1,8 +1,14 @@
+import os
 import subprocess
 
 import pytest
 
 
+def normalize_output(output: str):
+    """Normalizes string output across OSes"""
+    return output.replace("\r\r", "\r").strip()
```
Comment (angela-tran) [inline tests/test_hashfields.py:9]: Should this be `\r\n`?

Comment (angela-tran) [reply tests/test_hashfields.py:9]: I see the tests did pass on Windows, so I guess this is working? But I thought `\r\n` was the end-of-line terminator for Windows and haven't seen `\r\r` before.

Comment (thekaveman) [reply tests/test_hashfields.py:9]: Yeah maybe I should have a comment, but it was only for tests so I didn't care that much..

This is coming from output captured from the `stdout` and `stderr`, there is an additional `\r` in front of the newline in Windows. You should be able to try this on your local Windows machine to see it (in a Git Bash terminal):

```bash
# activate virtual env
python -m venv .venv
source .venv/Scripts/activate

# install this lib
pip install -e .[test]
```

Now run the tests in VS Code with a breakpoint set on one of the calls to this `normalize_output()` from `capfd` -- you'll see the extra `\r` in there! Arrggg 🤯

File: tests/test_hashfields.py [hunk 2]
```diff
@@ -1,8 +1,14 @@
+import os
 import subprocess
 
 import pytest
 
 
+def normalize_output(output: str):
+    """Normalizes string output across OSes"""
```
Comment (thekaveman) [inline tests/test_hashfields.py:8]:
```suggestion
    """Normalizes string output across OSes by removing extra whitespace emitted in some stdout/stderr"""
```

Comment (thekaveman) [reply tests/test_hashfields.py:8]: @angela-tran

File: tests/test_hashfields.py (hunk 1 above)
Comment (angela-tran) [reply tests/test_hashfields.py:9]: Ah I see. Thanks!

**Review by angela-tran — DISMISSED** [review]

**Comment by thekaveman** [comment]
I'm going to update the Test workflow to run tests in all 3 environments so we see errors earlier (before making a release).

**Review by angela-tran — APPROVED** [review]

**Pull request merged**